cpp/unreal_plugins/CoreUtils/Source/CoreUtils/InputActionComponent.h

```diff
@@ -0,0 +1,113 @@
+//
+// Copyright(c) 2022 Intel. Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+//
+
+#pragma once
+
+#include <functional>
+#include <map>
+#include <string>
+#include <vector>
+
+#include <Components/InputComponent.h>
+#include <Components/SceneComponent.h>
+#include <Engine/EngineBaseTypes.h> // ELevelTick
+#include <GameFramework/PlayerController.h>
+#include <GameFramework/PlayerInput.h>
+#include <UObject/ObjectMacros.h>   // GENERATED_BODY, UCLASS
+
+#include "CoreUtils/Log.h"
+#include "CoreUtils/Unreal.h"
+
+#include "InputActionComponent.generated.h"
+
+struct FActorComponentTickFunction;
+
+UCLASS()
+class COREUTILS_API UInputActionComponent : public USceneComponent
+{
+    GENERATED_BODY()
+public:
+    UInputActionComponent()
+    {
+        SP_LOG_CURRENT_FUNCTION();
+
+        PrimaryComponentTick.bCanEverTick = true;
+        PrimaryComponentTick.bTickEvenWhenPaused = false;
+    }
+
+    virtual ~UInputActionComponent()
+    {
+        SP_LOG_CURRENT_FUNCTION();
+
+        input_component_ = nullptr;
+        input_action_descs_.clear();
+    }
+
+    // USceneComponent interface
+    void TickComponent(float delta_time, ELevelTick level_tick, FActorComponentTickFunction* this_tick_function) override
+    {
+        USceneComponent::TickComponent(delta_time, level_tick, this_tick_function);
+
+        if (input_component_ && apply_input_action_func_) {
+            for (auto& input_action_desc : input_action_descs_) {
+                float axis_value = input_component_->GetAxisValue(Unreal::toFName(input_action_desc.axis_));
+                if (axis_value >= input_action_desc.threshold_) {
+                    apply_input_action_func_(input_action_desc.key_);
+                }   
+            }
+        }
+    }
+
+    // Must be called in BeginPlay() or later as GetWorld() needs to be valid.
+    template <typename T>
+    void bindInputActions(const std::map<std::string, T>& input_actions)
+    {
+        SP_ASSERT(GetWorld());
+        SP_ASSERT(GetWorld()->GetFirstPlayerController());
+
+        for (auto& input_action : input_actions) {
```

Comment:
These two loops could be implemented more cleanly as a single loop.

Reply:
Made this code cleaner